src/lib_json/json_writer.cpp

```diff
@@ -118,21 +118,29 @@ JSONCPP_STRING valueToString(UInt value) {
 #endif // # if defined(JSON_HAS_INT64)
 
 namespace {
-JSONCPP_STRING valueToString(double value, bool useSpecialFloats, unsigned int precision) {
+JSONCPP_STRING valueToString(double value, bool useSpecialFloats, unsigned int precision, PrecisionType precisionType) {
   // Allocate a buffer that is more than large enough to store the 16 digits of
   // precision requested below.
   char buffer[36];
   int len = -1;
 
   char formatString[15];
-  snprintf(formatString, sizeof(formatString), "%%.%ug", precision);
+  if (precisionType == PrecisionType::significantDigits) {
+    snprintf(formatString, sizeof(formatString), "%%.%ug", precision);
```

Comment:
Shouldn't this be `e` instead of `g`?
* http://en.cppreference.com/w/cpp/io/c/fprintf

Reply:
I was thinking about saving old behaviour with `g` when used the shortest variant for float, because maybe some guys expect old behaviour from code. Maybe it is better to add third type for `e` ?

Reply:
You're right. Preserving the old way by default is a good idea.

I don't have an opinion on a third type.

Let's drop this latest commit. Then we can merge.

Reply:
Ok, i've dropped last commit. U can merge when test will be don.